cache/lru/lru.go

```diff
@@ -199,6 +236,11 @@ func (c *Cache[K, V]) LoadAndDelete(key K) (V, bool) {
 		return defaultVal, false
 	}
 
+	// Call the onEvict callback if set for the element.
+	c.onEvict.WhenSome(func(cb OnEvictCallback[K, V]) {
```

Comment:
Here onEvict is called during an intentional deletion (LoadAndDelete), which is not an eviction.